retrofit-converters/moshi/src/test/java/retrofit2/converter/moshi/MoshiConverterFactoryTest.java

```diff
@@ -269,4 +277,15 @@
     } catch (IOException expected) {
     }
   }
+
+  @Test public void requireFullConsumption() throws Exception {
+    server.enqueue(new MockResponse().setBody("{}[]"));
```

Comment:
Gson will normally fail already for this, since it does extra buffering, so I don't think Gson's converter needs an option for this.